pkg/apis/core/validation/validation_test.go

```diff
@@ -17341,7 +17342,7 @@ func TestValidateReplicationController(t *testing.T) {
 				}
 			}),
 			expectedErrs: field.ErrorList{
-				field.Invalid(field.NewPath("metadata.annotations"), nil, "name part must consist of"),
+				field.Invalid(field.NewPath("metadata.annotations"), nil, "").WithOrigin(""),
```

Comment:
Why not include a real origin here?

Reply:
The origin needs to be installed in a place which knows the intent.

`ValidateReplicationController()` -> `ValidateObjectMeta()` -> `apimachineryvalidation.ValidateObjectMeta()` -> `ValidateObjectMetaAccessor()` -> `ValidateAnnotations()` -> `validation.IsQualifiedName()` -- it seems like the right place to put an origin is in `ValidateAnnotations()` and the right origin is probably `k8s-label-key` (yes it is an annotation but the keys are the same format)